opus/application/static_media/js/widgets.js

```diff
@@ -803,14 +1175,194 @@ var o_widgets = {
+            if ($(singleDropdown).find("li").length > 0 && $(singleDropdown).is(":visible")) {
+                let slug = $(singleDropdown).attr("data-slug");
+                let slugWithCounter = $(singleDropdown).attr("data-input");
+
+                let inputPosition = $(`input[name="${slugWithCounter}"]`).offset();
+                let inputHeight = $(`input[name="${slugWithCounter}"]`).outerHeight();
+
+                let autocompletePos = {left: inputPosition.left, top: inputPosition.top + inputHeight};
+                $(singleDropdown).offset(autocompletePos);
+                $(singleDropdown).width($(`input[name="${slugWithCounter}"]`).width());
+            }
+        }
+    },
+
+    disableButtonsInAWidget: function(disable=true) {
+        /**
+         * Disable/enable "x", "+ (OR)" and trash icons in a widget.
+         */
+        if (disable) {
+            $(".op-remove-inputs").addClass("op-disable-btn");
+            $(".op-add-inputs").addClass("op-disable-btn");
+            $(".close_card").addClass("op-disable-btn");
+        } else {
+            $(".op-remove-inputs").removeClass("op-disable-btn");
+            $(".op-add-inputs").removeClass("op-disable-btn");
+            $(".close_card").removeClass("op-disable-btn");
+        }
+    },
+
+    destroyAutocomplete: function(slug) {
+        if ($(`#widget__${slug} input.STRING`).autocomplete().length !== 0) {
+            $(`#widget__${slug} input.STRING`).autocomplete("destroy");
+        }
+    },
+
+    initAutocomplete: function(slug, slugWithCounter) {
+        /**
+         * If we have a string input widget open, initialize autocomplete
+         * for string input
+         */
+
+        let displayDropDownList = true;
+
+        let stringInputDropDown = $(`#widget__${slug} input[name="${slugWithCounter}"].STRING`).autocomplete({
+            minLength: 1,
+            source: function(request, response) {
+                let currentValue = request.term;
+                let inputCounter = opus.getSlugOrDataTrailingCounterStr(slugWithCounter);
+                let idx = inputCounter ? parseInt(inputCounter)-1 : 0;
+
+                o_widgets.lastStringSearchRequestNo++;
+                o_search.slugStringSearchChoicesReqno[slugWithCounter] = o_widgets.lastStringSearchRequestNo;
+
+                if (opus.selections[slug]) {
+                    opus.selections[slug][idx] = currentValue;
+                } else {
+                    opus.selections[slug] = [currentValue];
+                }
+
+                let newHash = o_hash.updateHash(false);
+                /*
+                We are relying on URL order now to parse and get slugs before "&view" in the URL
+                Opus will rewrite the URL when a URL is pasted, all the search related slugs would be moved ahead of "&view"
+                Refer to hash.js getSelectionsFromHash and updateHash functions
+                */
+                let regexForHashWithSearchParams = /(.*)&view/;
```

Comment:
Actually I don't think this is true anymore, is it? view= appears near the end of the URL. I think cols= comes first now. This appears in multiple places throughout OPUS.